webui/react/src/components/NavigationSideBar.tsx

```diff
@@ -239,144 +242,150 @@ const NavigationSideBar: React.FC = () => {
+                      <Button
+                        icon={<Icon name="add-small" size="tiny" title="Create workspace" />}
+                        type="text"
                         onClick={WorkspaceCreateModal.open}
                       />
-                    </li>
-                  ) : workspaces.length === 0 ? (
-                    <div className={css.noWorkspaces}>No pinned workspaces</div>
-                  ) : null}
-                </ul>
-              ),
-              NotLoaded: () => <Spinner center />,
-            })}
-          </section>
-          <section className={css.bottom}>
-            {menuConfig.bottom.map((config) => (
-              <NavigationItem key={config.icon} tooltip={settings.navbarCollapsed} {...config} />
-            ))}
-            <NavigationItem
-              icon={settings.navbarCollapsed ? 'expand' : 'collapse'}
-              label={settings.navbarCollapsed ? 'Expand' : 'Collapse'}
-              tooltip={
-                settings.navbarCollapsed
-                  ? `Expand (${shortcutToString(navbarCollapsedShortcut)})`
-                  : `Collapse (${shortcutToString(navbarCollapsedShortcut)})`
-              }
-              onClick={handleCollapse}
-            />
+                    )}
+                  </div>
+                }
+                icon="workspaces"
+                key="workspaces"
+                label="Workspaces"
+                path={paths.workspaceList()}
+                tooltip={settings.navbarCollapsed}
+              />
+              {Loadable.match(pinnedWorkspaces, {
+                Loaded: (workspaces) => (
+                  <ul className={css.pinnedWorkspaces} role="list">
+                    {workspaces
+                      .sort((a, b) => ((a.pinnedAt ?? 0) < (b.pinnedAt ?? 0) ? -1 : 1))
+                      .map((workspace) => (
+                        <WorkspaceActionDropdown
+                          isContextMenu
+                          key={workspace.id}
+                          returnIndexOnDelete={false}
+                          workspace={workspace}>
+                          <li>
+                            <NavigationItem
+                              icon={<DynamicIcon name={workspace.name} size={24} />}
+                              label={workspace.name}
+                              labelRender={
+                                <Typography.Paragraph ellipsis={{ rows: 1, tooltip: true }}>
+                                  {workspace.name}
+                                </Typography.Paragraph>
+                              }
+                              path={paths.workspaceDetails(workspace.id)}
+                            />
+                          </li>
+                        </WorkspaceActionDropdown>
+                      ))}
+                    {workspaces.length === 0 && (
+                      <div className={css.noWorkspaces}>No pinned workspaces</div>
+                    )}
```

Comment:
not strictly related to the rest of this PR, but just noticed that this was only showing the "No pinned workspaces" message if `canCreateWorkspace` is `false`, which I assume is not the actual intended behavior. Also moving it above the "New workspace" link because that seems more appropriate for an empty message for the list above. 

Can separate this out from this PR if anyone feels strongly about that.